mindmeld/components/entity_resolver.py

```diff
@@ -542,43 +231,120 @@ def _process_entity_map(entity_type,
                 if augment_normalized and normalizer:
                     new_aliases.extend([normalizer(string) for string in aliases])
                 aliases = set([*aliases, *new_aliases])
-            if normalize_aliases:
-                alias_normalizer = normalizer
-                aliases = [alias_normalizer(alias) for alias in aliases]
+            if normalize_aliases and normalizer:
+                aliases = [normalizer(alias) for alias in aliases]
 
             items_for_cname = item_map.get(cname, [])
             items_for_cname.append(item)
             item_map[cname] = items_for_cname
             for alias in aliases:
                 if alias in syn_map:
                     msg = "Synonym %s specified in %s entity map multiple times"
-                    logger.debug(msg, cname, entity_type)
+                    logger.debug(msg, cname, self.type)
                 cnames_for_syn = syn_map.get(alias, [])
                 cnames_for_syn.append(cname)
                 syn_map[alias] = list(set(cnames_for_syn))
 
         return {"items": item_map, "synonyms": syn_map}
 
     def _load_entity_map(self, force_reload=False):
-        return self._resource_loader.get_entity_map(self.type, force_reload=force_reload)
+        try:
+            return self._resource_loader.get_entity_map(self.type, force_reload=force_reload)
+        except Exception as e:
+            msg = f"Unable to load entity mapping data for " \
+                  f"entity type: {self.type} in app_path: {self.app_path}"
+            raise Exception(msg) from e
 
-    @abstractmethod
-    def _fit(self, clean, entity_map):
-        """Fits the entity resolver model
+    @staticmethod
+    def _format_entity_map(entity_map):
+        if not entity_map.get("entities", []):
+            return entity_map
+
+        # format (if required) entities key-field
+        new_entities = []
+        id_counter = 0
+        for ent in entity_map.get("entities", []):
+            cname = ent.get("cname", None)
+            if not cname:
+                continue
+            _id, whitelist = ent.get("id", None), ent.get("whitelist", [])
+            if not _id:
+                _id = str(id_counter)
+                id_counter += 1
```

Comment:
does the id have some relative ordering? I ask since there might be instances when `ent.get("id")` is not None and return 42 while the next ent is None and the `id_counter ` is 42. In that case, we will have two entities with the same id. If ordering is not important, can we just gen a UUID?

Reply:
That's a good catch. I didn't realize IDs can be simple integer numbers. It is important that ids are unique and are not same across two docs. I'll make required changes to tackle this.

Reply:
Replaced this snippet with following in QA module as well
```python
            _id = doc.get("id", None)
            if _id in all_ids:
                msg = f"Found a duplicate id {_id} while processing {data_file}. "
                _id = uuid.uuid4()
                msg += f"Replacing it with a new id: {_id}"
                logger.warning(msg)
            if not _id:
                _id = uuid.uuid4()
                msg = f"Found an entry in {data_file} without a corresponding id. " \
                      f"Creating a random new id ({_id}) for this object."
                logger.warning(msg)
```